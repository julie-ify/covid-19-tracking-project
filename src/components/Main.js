import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import CountryHeader from './CountryHeader';
import Header from './Header';
import SelectCountry from './SelectCountry';

const Main = () => {
  let cardNumber = 0;
  const countries = useSelector((state) => state.countriesReducer.countries);
  const [searchBar, setSearchBar] = useState('');

  const handleChange = (e) => {
    setSearchBar(() => e.target.value);
  };

  return (
    <div className="appWrap">
      <Header />
      <div>
        <CountryHeader region={countries} />
        <div className="countrySearch">
          <div className="countries">stats by country - 2021</div>
          <input placeholder="Search..." className="searchBar" value={searchBar} onChange={handleChange} />
        </div>
        <ul className="mainGrid">
          {countries
            && countries
              .filter((item) => {
                if (searchBar === '') {
                  return item;
                }
                return item.name
                  .toLowerCase()
                  .includes(searchBar.toLocaleLowerCase());
              })
              .map((country) => {
                if (cardNumber + 1 > 4) {
                  cardNumber = 1;
                } else {
                  cardNumber += 1;
                }
                return (
                  <SelectCountry
                    country={country}
                    key={country.id}
                    cardNumber={cardNumber}
                  />
                );
              })}
        </ul>
      </div>
    </div>
  );
};

export default Main;
