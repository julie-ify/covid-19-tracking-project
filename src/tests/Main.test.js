import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/ConfigureStore';
import Main from '../components/Main';

it('renders correctly', () => {
  const main = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Main />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(main).toMatchSnapshot();
});
