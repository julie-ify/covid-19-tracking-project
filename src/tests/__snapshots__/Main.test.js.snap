// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="appWrap"
>
  <header
    className="appHeader"
  >
    <div
      className="headerFlex"
    >
      <div
        className="headerItems"
      >
        <a
          className="regionLink"
          exact="true"
          href="/"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-left fa-w-8 fontawesome"
            data-icon="angle-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 256 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
        <span>
          2021
        </span>
      </div>
      <div>
        
        <span>
          All Countries Data
        </span>
      </div>
      <div
        className="headerItems"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-microphone-alt fa-w-11 "
          data-icon="microphone-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 352 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M336 192h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16zM176 352c53.02 0 96-42.98 96-96h-85.33c-5.89 0-10.67-3.58-10.67-8v-16c0-4.42 4.78-8 10.67-8H272v-32h-85.33c-5.89 0-10.67-3.58-10.67-8v-16c0-4.42 4.78-8 10.67-8H272v-32h-85.33c-5.89 0-10.67-3.58-10.67-8v-16c0-4.42 4.78-8 10.67-8H272c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-cog fa-w-16 "
          data-icon="cog"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </header>
  <div>
    <div
      className="countryHeader"
    >
      <div
        className="imgWrap"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-globe-europe fa-w-16 imgCover"
          data-icon="globe-europe"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 496 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm200 248c0 22.5-3.9 44.2-10.8 64.4h-20.3c-4.3 0-8.4-1.7-11.4-4.8l-32-32.6c-4.5-4.6-4.5-12.1.1-16.7l12.5-12.5v-8.7c0-3-1.2-5.9-3.3-8l-9.4-9.4c-2.1-2.1-5-3.3-8-3.3h-16c-6.2 0-11.3-5.1-11.3-11.3 0-3 1.2-5.9 3.3-8l9.4-9.4c2.1-2.1 5-3.3 8-3.3h32c6.2 0 11.3-5.1 11.3-11.3v-9.4c0-6.2-5.1-11.3-11.3-11.3h-36.7c-8.8 0-16 7.2-16 16v4.5c0 6.9-4.4 13-10.9 15.2l-31.6 10.5c-3.3 1.1-5.5 4.1-5.5 7.6v2.2c0 4.4-3.6 8-8 8h-16c-4.4 0-8-3.6-8-8s-3.6-8-8-8H247c-3 0-5.8 1.7-7.2 4.4l-9.4 18.7c-2.7 5.4-8.2 8.8-14.3 8.8H194c-8.8 0-16-7.2-16-16V199c0-4.2 1.7-8.3 4.7-11.3l20.1-20.1c4.6-4.6 7.2-10.9 7.2-17.5 0-3.4 2.2-6.5 5.5-7.6l40-13.3c1.7-.6 3.2-1.5 4.4-2.7l26.8-26.8c2.1-2.1 3.3-5 3.3-8 0-6.2-5.1-11.3-11.3-11.3H258l-16 16v8c0 4.4-3.6 8-8 8h-16c-4.4 0-8-3.6-8-8v-20c0-2.5 1.2-4.9 3.2-6.4l28.9-21.7c1.9-.1 3.8-.3 5.7-.3C358.3 56 448 145.7 448 256zM130.1 149.1c0-3 1.2-5.9 3.3-8l25.4-25.4c2.1-2.1 5-3.3 8-3.3 6.2 0 11.3 5.1 11.3 11.3v16c0 3-1.2 5.9-3.3 8l-9.4 9.4c-2.1 2.1-5 3.3-8 3.3h-16c-6.2 0-11.3-5.1-11.3-11.3zm128 306.4v-7.1c0-8.8-7.2-16-16-16h-20.2c-10.8 0-26.7-5.3-35.4-11.8l-22.2-16.7c-11.5-8.6-18.2-22.1-18.2-36.4v-23.9c0-16 8.4-30.8 22.1-39l42.9-25.7c7.1-4.2 15.2-6.5 23.4-6.5h31.2c10.9 0 21.4 3.9 29.6 10.9l43.2 37.1h18.3c8.5 0 16.6 3.4 22.6 9.4l17.3 17.3c3.4 3.4 8.1 5.3 12.9 5.3H423c-32.4 58.9-93.8 99.5-164.9 103.1z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="headerTextWrap"
      >
        <div
          className="headerText"
        >
          <p
            className="bigHeading"
          >
            All Countries with Covid-19 cases
          </p>
          <span>
            0
             
            confirmed cases
          </span>
        </div>
      </div>
    </div>
    <div
      className="countrySearch"
    >
      <div
        className="countries"
      >
        stats by country - 2021
      </div>
      <input
        className="searchBar"
        onChange={[Function]}
        placeholder="Search..."
        value=""
      />
    </div>
    <ul
      className="mainGrid"
    />
  </div>
</div>
`;
